### DISCUSSION TITLE
resolving shortname

### DISCUSSION THREAD

**Discussion by gecko2308** [MichaCo/DnsClient.NET#109] [Q&A] [answered]
Hy,

I'm trying to figure how to resolve an asset whith his shortname in the local domain network.
full name: ec20dk123456.domain.local
short name: ec20dk123456

With the .net library from Microsoft, it can resolve without any issue.
But have no result with dnsclient...

I need to use dnsclient because I need to make the query on specific DNS server in the network.
Can you help to figure out how I can do that to get shortname resolve ?

Thanks !

**Comment by MichaConrad** [comment, marked as answer]
:update: see #118 

Hi @gecko2308,
This library currently only works with FQNs, a fully qualified domain name. Its a pure DNS library and will not query NetBios or WINS servers and all the other things the build in Windows DNS client does.
The .NET library System.Net.Dns uses the native Windows APIs which uses the Windows own Dns resolver.

You can figure out the domain suffix by scanning your network interfaces `NetworkInterface.GetAllNetworkInterfaces();` and look into `GetIPProperties();` of the right interface. If that's all configured you can append that suffix to the hostname your are trying to query.
But that might or might not work either depending on your network setup...
